microservices-connector/config/authN-authZ/README.md

```diff
@@ -79,5 +80,77 @@ kubectl exec -it -n chatqa $CLIENT_POD -- curl -X POST $accessUrl -d '{"text":"W
 
 # try with the valid token. Shall get the correct response
 kubectl exec -it -n chatqa $CLIENT_POD -- curl -X POST $accessUrl -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -sS -H 'Authorization: Bearer $TOKEN2' -H 'Content-Type: application/json'
+```
+
+## Perform authentication and authorization via JWT tokens generated by Keycloak
+
+Keycloak is an open source identity and access management project to add authentication to applications and secure services with minimum effort. Here we leverage Keycloak to generate a JWT token.
+
+**Keycloak installation and configuration**
+
+Install Keycloak in the kubernetes cluster for user management. **Note:** Replace the admin password as your own in the keycloak_install.yaml file.
+
+```bash
+cd $(pwd)
+kubectl apply -f $(pwd)/config/authN-authZ/keycloak_install.yaml
+
+# get the ip and port to access keycloak.
+export HOST_IP=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}' | cut -d '/' -f3 | cut -d ':' -f1)
+export KEYCLOAK_PORT=$(kubectl get svc keycloak -o jsonpath='{.spec.ports[0].nodePort}')
+export KEYCLOAK_ADDR=${HOST_IP}:${KEYCLOAK_PORT}
+```
+
+**Note:** Double check if the host ip captured is the correct ip.
+
+Access the Keycloak admin console through the `KEYCLOAK_ADDR` to configure the users. Use the username and password specified in the yaml file to login. Configuration steps look like this:
+
+1. Create a new realm named `istio` within Keycloak.
+
+2. Create a new client called `istio` with default configurations.
+
+3. From the left pane select the Realm roles and create a new role name as `user` and another new role as `viewer`.
+
+4. Create a new user name as `users` and another user as `userb`. Set passwords for both users (set 'Temporary' to 'Off'). Select Role mapping on the top, assign the `user` role to `users` and assign the `viewer` role to `userb`.
```

Comment:
Can we give another more meaningfull user name add some explanations for these two users?  and explain the scenario below.

Reply:
I changed the username to 'mary' and 'bob' in the case and add a simple explanation under the section "Perform authentication and authorization via JWT tokens generated by Keycloak"